packages/make-styles/src/makeStyles.ts

```diff
@@ -1,42 +1,81 @@
 import { DEFINITION_LOOKUP_TABLE, SEQUENCE_PREFIX } from './constants';
 import { createCSSVariablesProxy, resolveStyleRules } from './runtime/index';
 import { hashString } from './runtime/utils/hashString';
-import { MakeStylesOptions, MakeStylesStyleFunctionRule, MakeStylesStyleRule } from './types';
+import {
+  MakeStylesOptions,
+  MakeStylesRenderer,
+  MakeStylesResolvedRule,
+  MakeStylesStyleFunctionRule,
+  MakeStylesStyleRule,
+} from './types';
+
+type Created<Slots extends string> = Record<Slots, Record<string, MakeStylesResolvedRule>>;
+
+function resolveClasses<Slots extends string>(
+  resolvedStyles: Created<Slots>,
+  dir: 'ltr' | 'rtl',
+  renderer: MakeStylesRenderer,
+) {
+  const resolvedClasses = {} as Record<Slots, string>;
+
+  // eslint-disable-next-line guard-for-in
+  for (const slotName in resolvedStyles) {
+    const slotClasses = renderer.insertDefinitions(dir, resolvedStyles[slotName]);
```

Comment:
It is `renderer`'s responsibility to compute the classnames, but the rest of the code expect the classnames are stable and atomic. Where do we have this contract described and tested?

Reply:
It is not described. For me it's unclear path yet...


- Will we implement CSS extraction to .css files?
- How will look build time transform?
- Where is the balance between memory and perf for runtime/build time?

I don't have answers and thus don't want to make a strict contract there yet...